core/runtime/runtime.h

```diff
@@ -11,6 +11,73 @@ namespace runtime {
 
 using EngineID = int64_t;
 
+typedef enum { DeviceIdx = 0, EngineIdx } SerializedInfoIndex;
+
+struct CudaDevice {
+  int64_t id; // CUDA device id
+  int64_t major; // CUDA compute major version
+  int64_t minor; // CUDA compute minor version
+  nvinfer1::DeviceType device_type;
+  size_t device_name_len;
```

Comment:
Why do you need this?

Reply:
I need to know the byte size I am supposed to deserialize for the device_name field.

Reply:
Can you just get the size of the string?

Reply:
how will that work during deserialization? It needs to understand how many bytes form part of device_name.  For other primitive data types, it makes sense as it will be parsing based on sizeof(datatype) but for string it will vary, I don't think we have such an option.

Reply:
Resolving this conversation as per discussion.